model_compression_toolkit/core/common/mixed_precision/resource_utilization_tools/resource_utilization_calculator.py

```diff
@@ -0,0 +1,646 @@
+        """
+        Compute graph's weights resource utilization.
+
+        Args:
+            target_criterion: criterion to include targets for computation.
+            bitwidth_mode: bit-width mode for computation.
+            w_qcs: weights quantization config per node for the custom bit mode. Must provide configuration for all
+              configurable weights.
+
+        Returns:
+            - Total weights utilization.
+            - Per node total utilization. Dict keys are nodes in a topological order.
+            - Detailed per node per weight utilization. Dict keys are nodes in a topological order.
+        """
+        nodes = self._get_target_weight_nodes(target_criterion, include_reused=False)
+
+        util_per_node: Dict[BaseNode, Utilization] = {}
+        util_per_node_per_weight = {}
+
+        for n in self._topo_sort(nodes):
+            w_qc = w_qcs.get(n) if w_qcs else None
+            node_weights_util, per_weight_util = self.compute_node_weights_utilization(n, target_criterion,
+                                                                                       bitwidth_mode, w_qc)
+            util_per_node[n] = node_weights_util
+            util_per_node_per_weight[n] = per_weight_util
+
+        aggregate_fn = ru_target_aggregation_fn[RUTarget.WEIGHTS]
+        total_util = aggregate_fn(u.by_bit_mode(bitwidth_mode) for u in util_per_node.values())
+        return total_util, util_per_node, util_per_node_per_weight
+
+    def compute_node_weights_utilization(self,
+                                         n: BaseNode,
+                                         target_criterion: TargetInclusionCriterion,
+                                         bitwidth_mode: BitwidthMode,
+                                         qc: NodeWeightsQuantizationConfig)\
+            -> Tuple[Utilization, Dict[str, Utilization]]:
+        """
+        Compute resource utilization for weights of a node.
+
+        Args:
+            n: node.
+            target_criterion: criterion to include weights for computation.
+            bitwidth_mode: bit-width mode for the computation.
+            qc: weight quantization config for the custom bit mode computation. Must provide configuration for all
+              configurable weights.
+
+        Returns:
+            - Total utilization.
+            - Detailed per weight utilization.
+        """
+        weight_attrs = self._get_target_weight_attrs(n, target_criterion)
+        if not weight_attrs:    # pragma: no cover
+            return Utilization.zero_utilization(bitwidth_mode, ), {}
+
+        attr_util = {}
+        for attr in weight_attrs:
+            size = self._params_cnt[n][attr]
+            bytes_ = None
+            if bitwidth_mode != BitwidthMode.Size:
+                nbits = self._get_weight_nbits(n, attr, bitwidth_mode, qc)
+                bytes_ = size * nbits / 8
```

Comment:
consider adding a "size_to_byte" method, I think it will be clearer to understand the exact computation that we are doing when we stumble into this in the future (also - reuse in all targets' computation)